symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py

```diff
@@ -39,7 +43,48 @@
     molecules_volume_parameter - bonding_forces_parameter / (units.molar_gas_constant * temperature)
 )
 
-# TODO: Derive from the van der Waals equation of state
+# Derive from the van der Waals equation of state and the virial equation
+
+_volume = compressibility_def.volume
+_mole_count = compressibility_def.amount_of_substance
+_compressibility_sym = compressibility_def.compressibility_factor
+_molar_density = virial_eqn.molar_density
+
+_pressure_expr = solve(vdw_eqn.law, vdw_eqn.pressure)[0].subs({
+    vdw_eqn.volume: _volume,
+    vdw_eqn.temperature: temperature,
+    vdw_eqn.amount_of_substance: _mole_count,
+    vdw_eqn.bonding_forces_parameter: bonding_forces_parameter,
+    vdw_eqn.molecules_volume_parameter: molecules_volume_parameter,
+})
+
+_compressibility_via_volume = compressibility_def.definition.rhs.subs({
+    compressibility_def.pressure: _pressure_expr,
+    compressibility_def.temperature: temperature,
+})
+
+_compressibility_via_density = solve(
+    [
+        Eq(_compressibility_sym, _compressibility_via_volume),
+        Eq(_molar_density, _mole_count / _volume),
```

Comment:
Might be a good idea to also generalize it as some `volumetric_density` of something law.